core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1900,7 +1900,20 @@ private[spark] object Utils extends Logging {
    */
   def resolveURI(path: String): URI = {
     try {
-      val uri = new URI(path)
+      val osSafePath = if (Path.isWindowsAbsolutePath(path, false)) {
+        // Make sure C:/ part becomes /C/.
+        val windowsUri = new URI(path)
+        val driveLetter = windowsUri.getScheme
+        s"/$driveLetter/${windowsUri.getSchemeSpecificPart()}"
+      } else if (Path.isWindowsAbsolutePath(path, true)) {
+        // Make sure /C:/ part becomes /C/.
+        val windowsUri = new URI(path.substring(1))
+        val driveLetter = windowsUri.getScheme
+        s"/$driveLetter/${windowsUri.getSchemeSpecificPart()}"
```

Comment:
BTW if we can get to the test errors in https://gist.github.com/HyukjinKwon/0c42b2c208e06c59525d91087252d9b0 it will be very cool. The only remaining errors are from the LDA tests and that is about not locating the test data file -- that seems like a different bug that we can fix in a follow up